### PR TITLE
Practice7

### PR DESCRIPTION
**Pull request by stratopedarx**

### CODE REVIEW

**Review by vasilevkin — COMMENTED** [review]
Good start! Continue.

File: practice_2/GreatDayApp/Extensions/StringExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Foundation
+
+extension String {
+    func matches(_ regex: String) -> Bool {
+        return self.range(of: regex, options: .regularExpression, range: nil, locale: nil) != nil
```
Comment (vasilevkin) [inline practice_2/GreatDayApp/Extensions/StringExtensions.swift:5]: try do not compare to nil if possible

**Review by vasilevkin — APPROVED** [review]

File: practice_2/GreatDayApp/Extensions/StringExtensions.swift (hunk 1 above)
Comment (stratopedarx) [reply practice_2/GreatDayApp/Extensions/StringExtensions.swift:5]: If I do it like below, I get a swiftlint warning: Unused Optional Binding Violation: Prefer `!= nil` over `let _ =` 
This is the reason why I did it this way. Perhaps there are still ways to handle this point. I would be glad if you tell me. Thanks.
`
    func matches(_ regex: String) -> Bool {

        if let _ = self.range(of: regex, options: .regularExpression, range: nil, locale: nil) {
            return true
        }
        return false
    }
`

**Comment by stratopedarx** [comment]
@vasilevkin, prepared a small final project.
Please have a look.

File: practice_2/GreatDayApp/Extensions/StringExtensions.swift (hunk 1 above)
Comment (vasilevkin) [reply practice_2/GreatDayApp/Extensions/StringExtensions.swift:5]: LGTM, leave it as is
but some thoughts are here:
https://forums.swift.org/t/a-more-swifty-way-to-check-an-optional-for-a-nil-value/19221

**Pull request merged**